app/controllers/api/v0/serialisers.rb

```diff
@@ -101,19 +101,27 @@ def address_block(office, include_local_authority:)
 
       def opening_time_block(office, time_type)
         days = %w[Monday Tuesday Wednesday Thursday Friday Saturday Sunday]
-        open_days = days.reject { |day| office["#{time_type}_hours_#{day.downcase}"].nil? }
+        case time_type
+        when :office
+          opening_times = office.opening_hours
+          notes = office.opening_hours_information
+        when :telephone
+          opening_times = office.telephone_advice_hours
+          notes = office.telephone_advice_hours_information
+        end
+        open_days = days.reject { |day| opening_times[day.downcase.to_sym].nil? }
         open_days.map do |day|
-          opening_time_range = office["#{time_type}_hours_#{day.downcase}"]
-          if opening_time_range.nil?
+          opening_time_range = opening_times[day.downcase.to_sym]
+          if opening_time_range.empty?
             nil
           else
             {
               day:,
-              start1: opening_time_range.beginning.strftime("%H.%M"),
-              end1: opening_time_range.ending.strftime("%H.%M"),
-              start2: nil,
-              end2: nil,
-              notes: office["#{time_type}_hours_information"]
+              start1: opening_time_range[0].beginning.strftime("%H.%M"),
+              end1: opening_time_range[0].ending.strftime("%H.%M"),
+              start2: opening_time_range[1]&.beginning&.strftime("%H.%M"),
```

Comment:
Does this mean there are at most two sets of opening hours per day..?

Reply:
In Salesforce it's unbound. I suspect in Resource Directory it was limited to 2 sets, and because changing Episerver isn't really in scope (And this is the backwards-compatible API for Episerver) we're slightly constrained here. But the examples I've been made aware of are only 2 sets